misc.lua

```diff
@@ -4,9 +4,9 @@
 
 function BrewmasterTools.GetNextTick(unit)
   unit = UnitExists(unit) and unit or 'player'
-	return select(17,UnitDebuff(unit,(GetSpellInfo(124275))))
-	or select(17,UnitDebuff(unit,(GetSpellInfo(124274))))
-	or select(17,UnitDebuff(unit,(GetSpellInfo(124273))))
+	return select(16,BrewmasterTools.util.UnitDebuff(unit,(GetSpellInfo(124275))))
+	or select(16,BrewmasterTools.util.UnitDebuff(unit,(GetSpellInfo(124274))))
+	or select(16,BrewmasterTools.util.UnitDebuff(unit,(GetSpellInfo(124273))))
```

Comment:
This can be greatly simplified. The utility function  you build essentially loops through the buff tray until it finds the aura with the requested name. There's 2 problems with using this as a drop in replacement for finding the stagger debuff.

First, any of three debuffs could count. So, you want to look for a debuff that matches one of Light Stagger, Moderate Stagger, or Heavy Stagger. Looping through the buff tray 3 times will do this of course, but it is better to just do the loop once, and check each debuff against all three possibilities.

Second, GetSpellInfo was included because the names Light/Moderate/Heavy Stagger are localized names, and the old UnitAura method did not support spellid queries (the new one still doesn't, as it happens). But when you are querying via the buff tray, then you can just look to see if the buff id matches the right spell id.

Reply:
Thanks for the feedback. Realised after I posted the pull request that it could have used a bit more optimisation. Updated it with a better version.